src/gc.rs

```diff
@@ -0,0 +1,91 @@
+//! Utilities for performing garbage collection.
+#![allow(clippy::implicit_hasher)]
+use crate::block::decode_ipld;
+use crate::cid::Cid;
+use crate::error::Error;
+use crate::ipld::Ipld;
+use crate::store::ReadonlyStore;
+use std::collections::HashSet;
+
+/// Returns the references in an ipld block.
+pub fn references(ipld: &Ipld) -> HashSet<Cid> {
+    let mut set: HashSet<Cid> = Default::default();
+    for ipld in ipld.iter() {
+        if let Ipld::Link(cid) = ipld {
+            set.insert(cid.to_owned());
+        }
+    }
+    set
+}
+
+/// Returns the recursive references of an ipld block.
```

Comment:
This might be a non-native speaker (me) thing. I don't associate anything with "recursive references". Reading the code, it looks like you are traversing a DAG starting at some CID and returning all the references that you've visisted.

What about something like:

    /// Traverses a DAG starting at the `roots` and returns all the links it encounters

Reply:
A nitpick (something that I would change, but doesn't block merging as it currently is): I find `closure()` a non descriptive name. What about something like `recursive_references()` or `all_references()`.